lib/ansible/modules/network/cumulus/nclu.py

```diff
@@ -0,0 +1,193 @@
+        - add int swp1
+    atomic: true
+    description: "Ansible - add swp1"
+'''
+
+RETURN = '''
+changed:
+    description: whether the interface was changed
+    returned: changed
+    type: bool
+    sample: True
+msg:
+    description: human-readable report of success or failure
+    returned: always
+    type: string
+    sample: "interface bond0 config updated"
+'''
+
+
+def command_helper(module, command, errmsg=None):
+    """Run a command, catch any nclu errors"""
+    (_rc, output, _err) = module.run_command("/usr/bin/net %s"%command)
+    if _rc or 'ERROR' in output or 'ERROR' in _err:
+        module.fail_json(msg=errmsg or output)
+    return str(output)
+
+
+def check_pending(module):
+    """Check the pending diff of the nclu buffer."""
+    pending = command_helper(module, "pending", "Error in pending config. You may want to view `net pending` on this target.")
+
+    delimeter1 = "net add/del commands since the last 'net commit'"
+    color1 = '\x1b[94m'
+    if delimeter1 in pending:
+        pending = pending.split(delimeter1)[0]
+        pending = pending.replace('\x1b[94m', '')
+    return pending.strip()
+
+
+def run_nclu(module, command_list, command_string, commit, atomic, abort, description):
+    _changed = False
+
+    commands = []
+    if command_list:
+        commands = command_list
+    elif command_string:
+        commands = command_string.splitlines()
+
+    do_commit = False
+    do_abort = abort
+    if commit or atomic:
+        do_commit = True
+        if atomic:
+            do_abort = True
+
+    if do_abort:
+        command_helper(module, "abort")
+
+    # First, look at the staged commands.
+    before = check_pending(module)
+    # Run all of the the net commands
+    output_lines = []
+    for line in commands:
+        output_lines += [command_helper(module, line.strip(), "Failed on line %s"%line)]
+    output = "\n".join(output_lines)
+
+    # If pending changes changed, report a change.
+    after = check_pending(module)
+    if before == after:
+        _changed = False
+    else:
+        _changed = True
+
+    # Do the commit.
+    if do_commit:
+        result = command_helper(module, "commit description '%s'"%description)
+        if "commit ignored" in result:
+            _changed = False
+            command_helper(module, "abort")
+        elif command_helper(module, "show commit last") == "":
+            _changed = False
+
+    return _changed, output
+
+
+def main(testing=False):
+    module = AnsibleModule(argument_spec=dict(
+        commands = dict(required=False, type='list'),
+        template = dict(required=False, type='str'),
+        description = dict(required=False, type='str', default="Ansible-originated commit"),
+        abort = dict(required=False, type='bool', default=False),
+        commit = dict(required=False, type='bool', default=False),
+        atomic = dict(required=False, type='bool', default=False)),
+        mutually_exclusive=[('commands', 'template'),
```

Comment:
The docs should state this

"Mutually exclusive options must be documented as the final sentence on each of the options."

http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#examples-block